docs/guides/universal-receiver-delegate/create-custom-urd-2.md

```diff
@@ -0,0 +1,294 @@
+    1,
+    ethers.ZeroAddress,
+    0,
+    fullBytecode,
+  );
+  await tx1.wait();
+
+  console.log(
+    '✅ Custom URD successfully deployed at address: ',
+    CustomURDAddress,
+  );
+
+  // -------------------------
+  // GRANT URD PERM - METHOD 1
+  // -------------------------
+  // COMMENT IF YOU USE METHOD 2
+
+  // we need the key to store our custom URD contract address
+  // {_LSP1_UNIVERSAL_RECEIVER_DELEGATE_PREFIX + <bytes32 typeId>}
+  console.log('⏳ Registering custom URD on the UP');
+  const URDdataKey =
+    ERC725YDataKeys.LSP1.LSP1UniversalReceiverDelegatePrefix +
+    LSP1_TYPE_IDS.LSP7Tokens_RecipientNotification.slice(2).slice(0, 40);
+
+  // we will update the keys for:
+  // - the custom URD of specific TYPE_ID (with our custom URD contract address)
+  // - the permission of this custom URD contract (this will create a new controller in the Browser Extension)
+  const dataKeys = [
+    URDdataKey,
+    ERC725YDataKeys.LSP6['AddressPermissions:Permissions'] +
+      CustomURDAddress.slice(2),
+  ];
+
+  // Calculate the correct permission (SUPER_CALL + REENTRANCY)
+  const permInt =
+    parseInt(PERMISSIONS.SUPER_CALL, 16) ^ parseInt(PERMISSIONS.REENTRANCY, 16);
+  const permHex = '0x' + permInt.toString(16).padStart(64, '0');
+  const dataValues = [CustomURDAddress, permHex];
+
+  // execute the tx
+  const setDataBatchTx = await UP.connect(signer).setDataBatch(
+    dataKeys,
+    dataValues,
+  );
+  await setDataBatchTx.wait();
+  console.log('✅ Custom URD has been correctly registered on the UP');
+
+  // --------------------------------------------
+  // REGISTER URD + AUTHORIZE OPERATOR - METHOD 2
+  // --------------------------------------------
+  // COMMENT IF YOU USE METHOD 2
+
+  // we need the key to store our custom URD contract address
+  // {_LSP1_UNIVERSAL_RECEIVER_DELEGATE_PREFIX + <bytes32 typeId>}
+  console.log('⏳ Registering custom URD on the UP');
+  const URDdataKey =
+    ERC725YDataKeys.LSP1.LSP1UniversalReceiverDelegatePrefix +
+    LSP1_TYPE_IDS.LSP7Tokens_RecipientNotification.slice(2).slice(0, 40);
+
+  // execute the tx
+  const setDataTx = await UP.connect(signer).setData(
+    URDdataKey,
+    CustomURDAddress,
+  );
+  await setDataTx.wait();
+  console.log('✅ Custom URD has been correctly registered on the UP');
+
+  console.log('⏳ Authorizing URD on Custom Token');
+  // we only authorize the first contract in the contractsAddr array, but feel free to add a loop :)
+  const CustomToken = await ethers.getContractAt(
+    'CustomToken',
+    contractsAddr[0] as string,
+  );
+
+  // Create the function call by encoding the function to be called and the params
+  const authBytes = CustomToken.interface.encodeFunctionData(
+    'authorizeOperator',
+    [CustomURDAddress, ethers.MaxUint256, '0x'], // we authorize CustomURDAddress to spend ethers.MaxUint256
+  );
+  // Execute the function call as the UP
+  const authTxWithBytes = await UP.connect(signer).execute(
+    0,
```

Comment:
same here, let's use the constants for better readability

```suggestion
    OPERATIONS_TYPE.CALL,
```

Reply:
done